idaes/core/initialization/initializer_base.py

```diff
@@ -0,0 +1,375 @@
+                f"initialization (DoF = {degrees_of_freedom(model)})."
+            )
+
+    def initialization_routine(self, model: Block):
+        """
+        Placeholder method to run initialization routine. Derived classes should overload
+        this with the desired routine.
+
+        Args:
+            model: Pyomo Block to initialize.
+
+        Returns:
+            Overloaded method should return a Pyomo solver results object is available,
+            otherwise None
+
+        Raises:
+            NotImplementedError
+        """
+        self.status = InitializationStatus.Error
+        raise NotImplementedError()
+
+    def restore_model_state(self, model: Block):
+        """
+        Restore model state to that stored in self.initial_state.
+
+        Args:
+            model: Pyomo Block ot restore state on.
+
+        Returns:
+            None
+
+        Raises:
+            ValueError if no initial state is stored.
+        """
+        if self.initial_state is not None:
+            from_json(model, sd=self.initial_state, wts=StoreState)
+        else:
+            self.status = InitializationStatus.Error
+            raise ValueError("No initial state stored.")
+
+    def postcheck(self, model: Block, results_obj: dict = None):
+        """
+        Check the model has been converged after initialization.
+
+        If a results_obj is provided, this will be checked using check_optimal_termination,
+        otherwise this will walk all constraints in the model and check that they are within
+        tolerance (set via the Initializer constraint_tolerance config argument).
+
+        Args:
+            model: model to be checked for convergence.
+            results_obj: Pyomo solver results dict (if applicable, default=None).
+
+        Returns:
+            InitialationStatus Enum
+        """
+        if results_obj is not None:
+            self.postcheck_summary = {
+                "solver_status": check_optimal_termination(results_obj)
+            }
+            if not self.postcheck_summary["solver_status"]:
+                # Final solver call did not return optimal
+                self.status = InitializationStatus.Failed
+                raise InitializationError(
+                    f"{model.name} failed to initialize successfully: solver did not return "
+                    "optimal termination. Please check the output logs for more information."
+                )
+        else:
+            # Need to manually check initialization
+            # First, check that all non-stale Vars have values
+            uninit_vars = []
+            for v in model.component_data_objects(Var, descend_into=True):
+                if v.value is None:
+                    uninit_vars.append(v)
+            # Next check for unconverged equality constraints
+            uninit_const = large_residuals_set(model, self.config.constraint_tolerance)
```

Comment:
It will check residuals for all active inequalities. My first reaction here would be that if the user wants to ignore inequalities during initialization they probably should be deactivated.